src/main/java/com/openlattice/data/DataApi.java

```diff
@@ -154,6 +155,18 @@ Integer replaceAssociationData(
     @DELETE( BASE + "/" + ENTITY_SET + "/" + SET_ID_PATH )
     Void clearEntitySet( @Path( ENTITY_SET_ID ) UUID entitySetId );
 
+    /**
+     * Clears the Entity matching the given Entity id and all of its neighbor Entities
+     *
+     * @param vertexEntitySetId the id of the EntitySet to delete from
+     * @param vertexEntityKeyId the id of the Entity to delete
+     */
+    @DELETE( BASE + "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/" + NEIGHBORS )
```

Comment:
I think it using a query parameter instead of a path would be a better pattern here.